collector.go

```diff
@@ -109,6 +109,9 @@ func ScrapeTarget(ctx context.Context, target string, config *config.Module, log
 	// Do the actual walk.
 	err := snmp.Connect()
 	if err != nil {
+		if err == context.Canceled {
+			return nil, fmt.Errorf("timeout connecting to target %s", target)
```

Comment:
This is incorrect, it's the scrape request's tcp connection being closed - not a timeout.

Reply:
Ahh, ok.

Reply:
You're still using the word timeout. Any timeout is on the Prometheus side, not in the SNMP exporter.

Reply:
How about `scrape canceled (possible timeout) ...`?

Reply:
That could work